src/main/java/me/crespel/karaplan/service/impl/PlaylistServiceImpl.java

```diff
@@ -285,6 +285,19 @@ public Playlist sort(Playlist playlist, PlaylistSortType sortType, PlaylistSortD
 		return save(playlist, user);
 	}
 
+	@Override
+	public Playlist changeOrder(Playlist playlist, List<Long> songIdList) {
+		for(int i = 0; i < songIdList.size(); i++) {
+			Song s = songRepo.findById(songIdList.get(i)).get();
+			for(PlaylistSong ps : playlist.getSongs()) {
+				if(ps.getSong().getId().equals(s.getId())) {
+					ps.setPosition(i);
+				}
+			}
+		}
+		return playlistRepo.save(playlist);

```

Comment:
Use the `save(Playlist, User)` method from this service (which also checks if user is a member of the playlist) instead of calling the repo directly.